File: src/main/java/com/example/entity/ErrorEntity.java
package com.example.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.http.HttpStatus;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ErrorEntity {
    private HttpStatus status;
    private List<String> details = new ArrayList<>();

    public int getStatus() {
        return status.value();
    }

    public String getMessage() {
        return status.getReasonPhrase();
    }

    public void addDetail(String message) {
        details.add(message);
    }
}

File: src/main/java/com/example/entity/ErrorEntity.kt
package com.example.entity

import org.springframework.http.HttpStatus
import java.util.*

data class ErrorEntity(val status: HttpStatus) {

    val details: MutableList<String> = ArrayList<String>()

    fun getStatus(): Int? {
        return status.value()
    }

    val message: String?
        get() = status.reasonPhrase

    fun addDetail(message: String) {
        details.add(message)
    }
}
